op-service/queue/queue.go

```diff
@@ -15,13 +15,12 @@ func (q *Queue[T]) Enqueue(t ...T) {
 // (if there is one) and returns it. Returns a zero value and false
 // if there is no element to dequeue.
 func (q *Queue[T]) Dequeue() (T, bool) {
-	if len(*q) == 0 {
+	el, ok := q.DequeueN(1)
+	if !ok {
 		var zeroValue T
 		return zeroValue, false
 	}
-	t := (*q)[0]
-	*q = (*q)[1:]
-	return t, true
+	return el[0], true
```

Comment:
Admittedly, the old impl was better for Dequeue 🙈 sorry, didn't realize that we have to deal with a slice.